### PR TITLE
[risk=low][RW-9006] Refactoring the runtime panel to prepare for the apps panel

### PR DESCRIPTION
**Pull request by jmthibault79**
Changes that will be needed for the new Apps Panel, split out from that work for easier review.

Does not satisfy the AC for RW-9006.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: ui/src/app/utils/runtime-utils.tsx [hunk 1]
```diff
@@ -870,14 +870,16 @@ const useRuntime = (currentWorkspaceNamespace) => {
...
-            workspaceNamespace: currentWorkspaceNamespace,
+        const currentStore = runtimeStore.get();
+        if (currentWorkspaceNamespace === currentStore.workspaceNamespace) {
+          const newStore = {
+            ...currentStore,
             runtime: leoRuntime,
             runtimeLoaded: true,
-          });
+          };
+          if (!fp.isEqual(currentStore, newStore)) {
```
Comment (jmthibault79) [inline ui/src/app/utils/runtime-utils.tsx:880]: this check substantially reduces how often this store is updated

**Comment by jmthibault79** [comment]
1 failed e2e test which I don't expect to be relevant: `tests/cohorts/cohort-review.spec.ts`

File: ui/src/app/utils/runtime-utils.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/utils/runtime-utils.tsx:880]: TODO move comment to code

**Review by dmohs — APPROVED** [review]
Very nice refactor.

**Review by evrii — APPROVED** [review]
Thanks for walking us through this. This was a great refactor.

File: ui/src/app/utils/runtime-utils.tsx (hunk 1 above)
Comment (jmthibault79) [reply ui/src/app/utils/runtime-utils.tsx:882]: done

**Pull request merged**